src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -759,79 +739,37 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
     return visitTypesafe(ctx);
   }
 
-  PropertyPath visitTypesafe(SparqlAutomaticParser::PathContext* ctx) {
-    return visitChildren(ctx).as<PropertyPath>();
-  }
+  PropertyPath visitTypesafe(SparqlAutomaticParser::PathContext* ctx);
 
   antlrcpp::Any visitPathAlternative(
       SparqlAutomaticParser::PathAlternativeContext* ctx) override {
     return visitTypesafe(ctx);
   }
 
   PropertyPath visitTypesafe(
-      SparqlAutomaticParser::PathAlternativeContext* ctx) {
-    auto paths = visitVector<PropertyPath>(ctx->pathSequence());
-
-    if (paths.size() == 1) {
-      return paths[0];
-    } else {
-      return PropertyPath::makeAlternative(std::move(paths));
-    }
-  }
+      SparqlAutomaticParser::PathAlternativeContext* ctx);
 
   antlrcpp::Any visitPathSequence(
       SparqlAutomaticParser::PathSequenceContext* ctx) override {
     return visitTypesafe(ctx);
   }
 
-  PropertyPath visitTypesafe(SparqlAutomaticParser::PathSequenceContext* ctx) {
-    auto paths = visitVector<PropertyPath>(ctx->pathEltOrInverse());
-
-    if (paths.size() == 1) {
-      return paths[0];
-    } else {
-      return PropertyPath::makeSequence(std::move(paths));
-    }
-  }
+  PropertyPath visitTypesafe(SparqlAutomaticParser::PathSequenceContext* ctx);
 
   antlrcpp::Any visitPathElt(
       SparqlAutomaticParser::PathEltContext* ctx) override {
     return visitTypesafe(ctx);
   }
 
-  PropertyPath visitTypesafe(SparqlAutomaticParser::PathEltContext* ctx) {
-    PropertyPath p = visitTypesafe(ctx->pathPrimary());
-
-    if (ctx->pathMod()) {
-      if (ctx->pathMod()->getText() == "+") {
-        p = PropertyPath::makeTransitiveMin(p, 1);
-      } else if (ctx->pathMod()->getText() == "?") {
-        p = PropertyPath::makeTransitiveMax(p, 1);
-      } else if (ctx->pathMod()->getText() == "*") {
-        p = PropertyPath::makeTransitive(p);
-      } else {
-        AD_FAIL()  // Should be unreachable.
-      }
-    }
-
-    return p;
-  }
+  PropertyPath visitTypesafe(SparqlAutomaticParser::PathEltContext* ctx);
```

Comment:
Okay, the moving of "everything" to the .cpp file should have been a separate PR...